libbeat/scripts/generate_fields_docs.py

```diff
@@ -82,36 +74,21 @@ def fields_to_asciidoc(input, output, beat):
         print "fields.yml file is empty. fields.asciidoc cannot be generated."
         return
 
-    # If no sections are defined, docs can't be generated
-    if "sections" not in docs.keys():
-        print "Sections is not defined in fields.yml. fields.asciidoc cannot be generated."
-        return
-
-    sections = docs["sections"]
-
-    # Add default section
-    sections["beat"] = "Beat"
-
-    # Check if sections is define
-    if sections is None:
-        print "No sections are defined in fields.yml. fields.asciidoc cannot be generated."
-        return
-
+    # Create sections from available fields
+    sections = {}
+    for v in docs["fields"]:
+        sections[v["key"]] = v["title"]
 
     for key in sorted(sections):
         output.write("* <<exported-fields-{}>>\n".format(key))
     output.write("\n")
 
-    for key in sorted(sections):
-        name = sections[key]
-
-        if key in docs:
-            section = docs[key]
-            if "type" in section:
-                if section["type"] == "group":
-                    section["name"] = name
-                    section["anchor"] = key
-                    document_fields(output, section, sections, "")
+    # Sort alphabetically by key
+    for v in sorted(docs["fields"], key=lambda field: field["key"]):
+        section = v
```

Comment:
Why not `for section in sorted(...`? I think that should be equivalent by simpler.